soc/board_specific_workflows/test_general.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python3
+# Copyright 2021 The CFU-Playground Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import general
+import unittest
+import unittest.mock
+
+
+class TestGeneralSoCWorkflow(unittest.TestCase):
+    """TestCase for the GeneralSoCWorkflow class.
+    
+    All methods except `run` are tested in this test case.  
+    """
+    def setUp(self):
+        """Sets up various mocks used throughout the tests."""
+
+        # Mocks the argparse arguments
+        self.mock_namespace = unittest.mock.MagicMock()
+        self.target_value = 'test_target'
+        self.target_keyword = 'target'
+        self.build_keyword = 'build'
+        self.build_value = True
+        self.call_kwargs = {
+            self.target_keyword: self.target_value,
+            self.build_keyword: self.build_value,
+        }
+        self.mock_namespace.configure_mock(**self.call_kwargs)
+
+        # Mocks the generated LiteXSoC
+        self.mock_soc = unittest.mock.MagicMock()
+        self.build_name = 'build_name'
+        self.mock_soc.configure_mock(build_name=self.build_name)
+
+        # Mocks the imported LiteX-Boards module
+        self.mock_module = unittest.mock.MagicMock()
+        self.mock_module.BaseSoC = unittest.mock.MagicMock(
+            return_value=self.mock_soc)
+
+        # Mocks the LiteX Builder
+        self.mock_builder = unittest.mock.MagicMock()
+        self.gateware_dir = 'test_gateware_dir/'
+        self.mock_builder.configure_mock(gateware_dir=self.gateware_dir)
+        self.mock_builder.build = unittest.mock.MagicMock()
+
+        # Overwrite the format_bistream_filename method to avoid os.
```

Comment:
I don't think it's necessary to avoid the os module in this case? The manipulation that format_bitstream_filename() does does not make external calls.

If it is necessary to avoid calls to "os"  please find a way to do that without monkey-patching. (Monkey-patching is neat, but it's a last-resort, and definitely not required here where you have complete control of the source of the function being called.)